core/src/Kokkos_Array.hpp

```diff
@@ -185,6 +196,10 @@ struct Array<T, 0> {
   // for default move constructor and move assignment operator.
   // Array( Array && ) = default ;
   // Array & operator = ( Array && ) = default ;
+
+ private:
+  friend KOKKOS_INLINE_FUNCTION constexpr void kokkos_swap(
+      Array<T, 0>&, Array<T, 0>&) noexcept(Impl::is_nothrow_swappable_v<T>) {}
```

Comment:
Why is this dependent on `is_nothrow_swappable<T>` and not just `noexcept(true)`?

It should have the same exception specification as [[array.zero]](https://eel.is/c++draft/array.zero#4).

Reply:
You're right, I had looked in cppreference and read the framed `noexcept` specifications and missed the next clause that clearly stated:

> For zero-sized arrays, [noexcept](https://en.cppreference.com/w/cpp/language/noexcept_spec) specification:  noexcept

Will fix.  Thanks.